grove/tomography/tomography.py

```diff
@@ -0,0 +1,727 @@
+        dimension = self.pauli_basis.ops[0].shape[0]
+        return (dimension * process_fidelity + 1.0) / (dimension + 1.0)
+
+    def to_super(self):
+        """
+        Compute the standard superoperator representation of the estimated process.
+
+        :return: The process as a superoperator.
+        :rytpe: qutip.Qobj
+        """
+        return self.sop
+
+    def to_choi(self):
+        """
+        Compute the choi matrix representation of the estimated process.
+
+        :return: The process as a choi-matrix.
+        :rytpe: qutip.Qobj
+        """
+        return qt.to_choi(self.sop)
+
+    def to_chi(self):
+        """
+        Compute the chi process matrix representation of the estimated process.
+
+        :return: The process as a chi-matrix.
+        :rytpe: qutip.Qobj
+        """
+        return qt.to_chi(self.sop)
+
+    def plot_pauli_transfer_matrix(self, ax):
+        """
+        Plot the elements of the Pauli transfer matrix.
+
+        :param matplotlib.Axes ax: A matplotlib Axes object to plot into.
+        """
+        title = "Estimated process"
+        ut.plot_pauli_transfer_matrix(self.r_est, ax, self.pauli_basis.labels, title)
+
+    def plot(self):
+        """
+        Visualize the process.
+
+        :return: The generated figure.
+        :rtype: matplotlib.Figure
+        """
+        fig, (ax1) = plt.subplots(1, 1, figsize=(10, 8))
+        self.plot_pauli_transfer_matrix(ax1)
+        return fig
+
+
+def bitstring_to_int(bitstring):
+    """Convert a binary bitstring into the corresponding unsigned integer.
+    """
+    ret = 0
+    for b in bitstring:
+        ret = (ret << 1) | (int(b) & 1)
+    return ret
+
+
+def sample_assignment_probs(qubits, nsamples, cxn):
+    num_qubits = len(qubits)
+    dimension = 2 ** num_qubits
+    hists = []
+    preps = ut.basis_state_preps(*qubits)
+    for jj, p in izip(TRANGE(dimension), preps):
+        results = cxn.run_and_measure(p, qubits, nsamples)
+        idxs = list(map(bitstring_to_int, results))
+        hists.append(ut.make_histogram(idxs, dimension))
+    return ut.estimate_assignment_probs(hists)
+
+
+def do_tomography(target_program, nsamples, cxn, qubits, max_num_qubits, tomography_class,
+                  program_generator, settings):
+    if qubits is None:
+        qubits = sorted(target_program.get_qubits())
+
+    num_qubits = len(qubits)
+    dimension = 2 ** num_qubits
+
+    if num_qubits > max_num_qubits:
+        raise ValueError("Too many qubits!")
+
+    assignment_probs = sample_assignment_probs(qubits, nsamples, cxn)
+    tomo_seq = list(program_generator(target_program, qubits))
+    histograms = np.zeros((len(tomo_seq), dimension))
+    for i, tomo_prog in izip(TRANGE(len(tomo_seq)), tomo_seq):
+        results = cxn.run_and_measure(tomo_prog, qubits, nsamples)
+        idxs = list(map(bitstring_to_int, results))
+        histograms[i] = ut.make_histogram(idxs, dimension)
```

Comment:
I haven't gotten to the utilities yet, but it strikes me as odd that we need to roll our own histogram functionality instead of using numpy.